framework/utils/random-string.ts

```diff
@@ -21,3 +21,7 @@ export function randomString(
   }
   return text;
 }
+
+export function randomLowercaseString(length: number): string {
```

Comment:
We will lose all naming conventions we agreed on. It would be nice to leave the naming to Hub. It gives us more info when a test goes wrong. Plus we have objects that cannot be named just whatever.

Reply:
It would be nice to use consistent e2e test resource naming across the platform.
Any resources starting with e2e_, we should be able to assume they were created with E2E testing.

Reply:
Please do not throw away naming we had there in favour of an absolutely random string. Hub agreed to do naming in specific way that helps us to find out where the object came from to the `describe`. We want something like `hub_e2e_[test_name]_[describe_name]_[random string]`. We can switch `hub` and `e2e` anytime there's a consensus on naming on the platform level that is documented. We also need to start the name of some objects with specific prefix like roles need to start with `galaxy.`. Framework should allow for flexible naming instead of cutting individual projects from it.

Reply:
> Framework should allow for flexible naming instead of cutting individual projects from it.
If you mean project as HUB, AWX, or EDA, we are a Platform UI and a Platform UI Team.

Naming resources with really long names makes the tests run slower as the name has to be typed in by cypress.

Trying to put the full test name and describe name in the resource seems like it could be problematic. In addition if a test fails, you can see the name right in the failure.

Having hub_ on the beginning of the name is kind of redundant. All hub resources are hub resources.

Having a common pattern for naming resources makes it consistent across testing in the platform. Makes it so that you can goto any resource view and search for "starts with" e2e. This also allows us to have scripts to cleanup e2e resources that are left over from testing. We should be cleaning up resources in tests, but if the test fails, it does not cleanup.

The current hub tests are not consistent with naming of resources. Having a function to generate that name makes it more consistent. `randomE2Ename()`. If we want to enhance that function to support a postfix on the end of that name, i'm all for it.

The support cypress functions are designed to be fully flexible allowing overriding any properties of the resource.
```
createHubRemoteRegistry() // uses a default generated name
```
or
```
createHubRemoteRegistry({
  name: "some_name_I_want_to_use"
})
```

This gives us common patterns for working with the custom cypress commands for working with resources. This allows developers to easily work on tests across the whole platform.